File: Problem_05/problem_5.erl
-module(problem_5).
-compile(export_all).

start()->
    lists:foldr(fun(X, Y) -> lcm(X, Y) end, 1, lists:seq(2, 21)).

gcd(A, B) when B == 0 ->
    A;
gcd(A, B) ->
    gcd(B, A rem B).

lcm(A, B)->
    A * B div gcd(A, B).

File: project-euler/Problem_05/problem_5.js
function gcd(a, b) {
    if( b == 0)
	return a;
    else
	return gcd(b, a % b);
}

function lcm(a, b) {
    return (a * b) / gcd(a, b);
}

function problem_5() {
    var result = 1;

    for (var i = 2; i <= 20; i++) {
	result = lcm(result, i);
    }

    return result;
}


document.write('<b>result: </b>' + problem_5());